src/main/java/uk/gov/pay/directdebit/events/api/DirectDebitEventsResponse.java

```diff
@@ -23,7 +24,9 @@
     private DirectDebitEventsPagination pagination;
     
     public DirectDebitEventsResponse(List<DirectDebitEvent> events, int page, int total, DirectDebitEventsPagination directDebitEventsPagination) {
-        this.events = events;
+        for (DirectDebitEvent event: events) {
+            this.events.add(new DirectDebitEventExternalView(event));
+        }
```

Comment:
Alternatively, you could remove the initialisation of the `events` field with a `new ArrayList<>()` and do this:

```java
this.events = events.stream().map(DirectDebitEventExternalView::new).collect(toList());
```